plugins/pi4j-plugin-linuxfs/src/main/java/com/pi4j/plugin/linuxfs/provider/spi/LinuxFsSpi.java

```diff
@@ -312,29 +331,63 @@ public int write(byte b) {
         return write(new byte[] {b}, 0, 1);
     }
 
+
+    /**
+     * {@inheritDoc}
+     * write
+     *
+     *   SPI_BUFFSIZ most often is set to 4096.  See initialize()
+     * This implementation can write blocks greater than 4096 byte
+     * 'however', read and understand how this is accomplished.
+     *
+     * A write of data no greater than 4096 bytes is accomplished with
+     * a single SPI write operation.
+     * So CE line low, write bytes, CE line high
+     *
+     * A write greater than 4096 bytes will be segmented to multiple writes
+     * each 4096  bytes in length, the last write is the MOD value.
+     * So, CE line low, write first 4096 bytes CE line high.  This
+     * pattern will repeat until the last bytes length MOD 4096 are written.
+     * This means multiple SPI transaction with you SPI device. If CE line
+     * toggling creates problems with your SPI device your application
+     * can use a vacant GPIO configured as an Output pin and your
+     * application keep the CE pin low during the duration of the call to
+     * spi,write.
+     *
+     * 
+     * @param data data array of bytes to be written
+     * @param offset offset in data buffer to start at
+     * @param length number of bytes to be written
+     * @return
+     */
     @Override
     public int write(byte[] data, int offset, int length) {
-        // We could reuse this buffer for all requests as long as it is large enough.
-        // For now we'll alloc/free a new one on each request.
-        PeerAccessibleMemory buf = new PeerAccessibleMemory(length);
-        buf.write(0, data, offset, length);
-
-        spi_ioc_transfer transfer = new spi_ioc_transfer();
-        transfer.tx_buf = buf.getPeer();
-        transfer.rx_buf = 0;
-        transfer.bits_per_word = BITS8;
-        transfer.speed_hz = config.baud();
-        transfer.delay_usecs = 0;
-        transfer.len = length;
-
-        int ret = libc.ioctl(fd, SPI_IOC_MESSAGE(1), transfer);
-        if (ret < 0) {
-            LOG.error("Could not write SPI message. ret {}, error: {}", ret, Native.getLastError());
-            length = 0;
+        Objects.checkFromIndexSize(offset, length, data.length);
+
+        int start = offset;
+        while (start < data.length) {
+            PeerAccessibleMemory buf = new PeerAccessibleMemory(SPI_BUFFSIZ);
+            spi_ioc_transfer txEntry = new spi_ioc_transfer() ;
+            int end = Math.min(data.length, start + SPI_BUFFSIZ);
+            byte[] chunk = Arrays.copyOfRange(data, start, end);
+            buf.write(0, chunk, 0 , chunk.length);
```

Comment:
Why the chunk allocation here -- opposed to 
```
buf.write(0, data, start, end - start);
```

Reply:
done